### ISSUE TITLE
Scope dependency is upgraded by major after install

### ISSUE THREAD

**Issue by huntercaron** [jspm/generator#165]
When installing `connectkit` using a newly created generator, one of the scope dependencies is bumped a major. 

<img width="433" alt="Screenshot 2022-08-11 at 15 33 30" src="https://user-images.githubusercontent.com/4625978/184224381-34f7bfa6-4c62-403c-8095-0b8a401ecdbc.png">

This does not happen when using the same generator instance between installs, but due to our worker setup at Framer we are not able to share the same generator instance, which leads to this bug.

**Repro** using `@jspm/generator` beta.35:
(This does not happen in beta.31)

```js 
import { Generator } from "@jspm/generator";

const RESOLUTIONS = {
  "@react-three/fiber": "7.0.27",
};

const PROVIDED_DEPENDENCIES = new Set([
  "react",
  "react/jsx-runtime",
  "react-dom",
  "react-dom/server",
  "framer",
  "framer-motion",
  "framer-motion/three",
]);

const BASE_CONFIG = {
  mapUrl: "about:blank",
  ignore: Array.from(PROVIDED_DEPENDENCIES),
  env: ["production", "browser", "module"],
  resolutions: RESOLUTIONS,
};

async function Repro() {
  const generatorOne = new Generator(BASE_CONFIG);

  await generatorOne.install("@react-three/fiber");
  console.log(generatorOne.getMap().scopes["https://ga.jspm.io/"].zustand);

  const generatorTwo = new Generator({
    ...BASE_CONFIG,
    inputMap: generatorOne.getMap(),
  });

  await generatorTwo.install("wagmi");
  console.log(generatorTwo.getMap().scopes["https://ga.jspm.io/"].zustand);

  const generatorThree = new Generator({
    ...BASE_CONFIG,
    inputMap: generatorTwo.getMap(),
  });

  await generatorThree.install("connectkit");
  console.log(generatorThree.getMap().scopes["https://ga.jspm.io/"].zustand);
}

Repro().catch((e) => console.error(e));
```

**Comment by guybedford** [comment]
I managed to track down the issue here, and it's an interesting edge case. Unfortunately won't get a fix together today, but we can definitely get one through by Monday / Tuesday next week.

**Comment by huntercaron** [comment]
amazing, thank you!

**Comment by guybedford** [comment]
Turned out to be a big and important reworking, thanks for sharing the case. Fix released in 1.0.0-beta.36.
